CivetWeb Support for STM32H7 Series

**Describe the bug**
Building `samples/net/sockets/civetweb` for `nucleo_h745zi_q` produces following warning:

```
warning: MINIMAL_LIBC_MALLOC_ARENA_SIZE (defined at lib/libc/Kconfig:92) was assigned the value
'131072' but got the value ''. Check these unsatisfied dependencies: MINIMAL_LIBC_MALLOC (=n),
MINIMAL_LIBC (=n). See
http://docs.zephyrproject.org/latest/reference/kconfig/CONFIG_MINIMAL_LIBC_MALLOC_ARENA_SIZE.html
and/or look up MINIMAL_LIBC_MALLOC_ARENA_SIZE in the menuconfig/guiconfig interface. The Application
Development Primer, Setting Configuration Values, and Kconfig - Tips and Best Practices sections of
the manual might be helpful too.
```

This warning leads to following error in example's compilation:
```
In file included from <your_path_here>/zephyrproject/zephyr/samples/net/sockets/civetweb/src/libc_extensions.c:9:
<your_path_here>/zephyrproject/zephyr/samples/net/sockets/civetweb/src/libc_extensions.h:42:5: error: expected identifier or '(' before 'int'
   42 | int ferror(FILE *stream);
```

**To Reproduce**
Steps to reproduce the behavior:
> west build -p auto -b "nucleo_h745zi_q_m7" "samples/net/sockets/civetweb"

**Expected behavior**
A clearly compiled and running project on mentioned board.

**Impact**
Non-compilable default sample for mentioned board

**Environment (please complete the following information):**
 - OS: Linux
 - Toolchain: Zephyr SDK v0.11.4
 - Commit SHA or Version used: 5328adcff5e110559f2b179318102e8700a7e172


**Additional context**
My investigations showed, that there is explicit `fulllibc` select in `stm32h7` series and no random generator support.
P.S.: The mentioned sample project can be perfectly compiled with `nucleo_f767zi` board.

**Possible solution**
1. Add `RNG` support to stm32h7 series ( can be solved by #27374 )
2. Remove explicit `fulllibc` select for stm32h7 series ( can be solved by #27389 )

--- a/boards/arm/nucleo_h745zi_q/nucleo_h745zi_q_m7.dts
+++ b/boards/arm/nucleo_h745zi_q/nucleo_h745zi_q_m7.dts
@@ -63,3 +63,7 @@
 &mac {
 	status = "okay";
 };
+
+&rng {
+	status = "okay";
+};

--- a/soc/arm/st_stm32/stm32h7/Kconfig.defconfig.series
+++ b/soc/arm/st_stm32/stm32h7/Kconfig.defconfig.series
@@ -16,4 +16,11 @@ config DMA_STM32_V1
 	default y
 	depends on DMA_STM32
 
+if ENTROPY_GENERATOR
+
+config ENTROPY_STM32_RNG
+	default y
+
+endif # ENTROPY_GENERATOR
+
 endif # SOC_SERIES_STM32H7X

--- a/soc/arm/st_stm32/stm32h7/soc.h
+++ b/soc/arm/st_stm32/stm32h7/soc.h
@@ -76,6 +76,10 @@
 #include <stm32h7xx_ll_tim.h>
 #endif /* CONFIG_PWM_STM32 */
 
+#ifdef CONFIG_ENTROPY_STM32_RNG
+#include <stm32h7xx_ll_rng.h>
+#endif /* CONFIG_ENTROPY_STM32_RNG */
+
 #endif /* !_ASMLANGUAGE */
 
 #endif /* _STM32F7_SOC_H7_ */

